src/vs/workbench/contrib/comments/browser/commentThreadAdditionalActions.ts

```diff
@@ -0,0 +1,96 @@
+
+import * as dom from 'vs/base/browser/dom';
+
+import { IAction } from 'vs/base/common/actions';
+import { IMenu, SubmenuItemAction } from 'vs/platform/actions/common/actions';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { MarshalledId } from 'vs/base/common/marshallingIds';
+import { IRange } from 'vs/editor/common/core/range';
+import * as languages from 'vs/editor/common/languages';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { CommentFormActions } from 'vs/workbench/contrib/comments/browser/commentFormActions';
+import { CommentMenus } from 'vs/workbench/contrib/comments/browser/commentMenus';
+import { ICellRange } from 'vs/workbench/contrib/notebook/common/notebookRange';
+import { IContextMenuService } from 'vs/platform/contextview/browser/contextView';
+
+export class CommentThreadAdditionalActions<T extends IRange | ICellRange> extends Disposable {
+	private _container: HTMLElement | null;
+	private _buttonBar: HTMLElement | null;
+	private _commentFormActions!: CommentFormActions;
+
+	constructor(
+		container: HTMLElement,
+		private _commentThread: languages.CommentThread<T>,
+		private _contextKeyService: IContextKeyService,
+		private _commentMenus: CommentMenus,
+		private _actionRunDelegate: (() => void) | null,
+		@IContextMenuService private contextMenuService: IContextMenuService,
+	) {
+		super();
+
+		this._container = dom.append(container, dom.$('.comment-additional-actions'));
+		dom.append(this._container, dom.$('.section-separator'));
+
+		this._buttonBar = dom.append(this._container, dom.$('.button-bar'));
+		this._createAdditionalActions(this._buttonBar);
+	}
+
+	private _showMenu() {
+		this._container?.classList.remove('hidden');
+	}
+
+	private _hideMenu() {
+		this._container?.classList.add('hidden');
+	}
+
+	private _enableDisableMenu(menu: IMenu) {
+		const groups = menu.getActions({ shouldForwardArgs: true });
+
+		// Show the menu if at least one action is enabled.
+		for (const group of groups) {
+			const [, actions] = group;
+			for (const action of actions) {
+				if (action.enabled) {
+					this._showMenu();
+					return;
+				}
+
+				for (const subAction of (action as SubmenuItemAction).actions ?? []) {
+					if (subAction.enabled) {
+						this._showMenu();
+						return;
+					}
+				}
+			}
+		}
+
+		this._hideMenu();
+	}
+
+
+	private _createAdditionalActions(container: HTMLElement) {
+		const menu = this._commentMenus.getCommentThreadAdditionalActions(this._contextKeyService);
+		this._register(menu);
+		this._register(menu.onDidChange(() => {
+			this._commentFormActions.setActions(menu);
+			this._enableDisableMenu(menu);
+		}));
+
+		this._commentFormActions = new CommentFormActions(container, async (action: IAction) => {
+			this._actionRunDelegate?.();
+
+			action.run({
+				thread: this._commentThread,
+				$mid: MarshalledId.CommentThreadReply
```

Comment:
If you want to use `MarshalledId.CommentThreadReply` here then there should also be a `text` property. Otherwise, it may make more sense to structure this as a `MarshalledId.CommentThread` and use that instead.

Reply:
I agree that CommentThread would make more sense, although that responds back only with a simplified reference to the CommentThread (ids) and not the actual thread object, so its not easily usable.
I would suggest that we add a CommentThreadInstance which would be able to retrieve the just the thread.
Otherwise we could fallback to the CommentThreadReply if you don't want to add the additional property to MarshalledIds.
What do you think?

Reply:
`CommentThreadInstance` works for me!